### PR TITLE
Bluetooth: controller: split: Port Enc setup to be queueable

### PR DESCRIPTION
**Pull request by cvinayak**
Port the fix for the controller implementation to make start
encryption queueable if there is any control procedure in
progress.

Refer to #15335.
Relates to #15335, #15186, #15958 and #14636.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

### CODE REVIEW

File: subsys/bluetooth/controller/ll_sw/ull_conn.c
Comment (joerchan) [inline subsys/bluetooth/controller/ll_sw/ull_conn.c]: move this pause_tx into llcp_enc?

**Review by joerchan — APPROVED** [review]
looks good, left a few small comments.

File: subsys/bluetooth/controller/ll_sw/ull_conn.c
Comment (cvinayak) [reply subsys/bluetooth/controller/ll_sw/ull_conn.c]: The state is maintained as a differential value:
0. Request lock is idle
1. Thread acquires a lock on the  request, if req == ack then req +=1;
2. Thread commits the request, req +=1 or (req-ack) == 2

If ISR occurs between the thread checking  for req == ack and req +=1, then in the thead the differential will be (req-ack) == 3, indicating a race condition. In this case, thread will back-off by reverting req -= 1, resulting in a final (req-ack) == 2 (i.e. ISR request supercedes).

On the contrary, if ISR detects that request is acquired it backs-off to retry later.

File: subsys/bluetooth/controller/ll_sw/ull_conn.c
Comment (cvinayak) [reply subsys/bluetooth/controller/ll_sw/ull_conn.c]: agreed. Will give a try.

**Review by carlescufi — APPROVED** [review]

**Comment by cvinayak** [comment]
We wait for review from @thoh-ot

**Review by thoh-ot — APPROVED** [review]
LGTM

**Pull request merged**
